internal/requestid/requestid_test.go

```diff
@@ -0,0 +1,101 @@
+package requestid
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func newRequest(t *testing.T) *http.Request {
+	t.Helper()
+	r, err := http.NewRequest(http.MethodGet, "https://example.com", http.NoBody)
+	require.NoError(t, err)
+	return r
+}
+
+func Test_Middleware(t *testing.T) {
+	requestWithID := newRequest(t)
```

Comment:
It'd be great if you'd add the occasional blank line here and there in this block. It'd make it easier to read. As an example, you can group `requestWithID` statements from `requestWithoutID` statements, etc. Whitespace is really strong!